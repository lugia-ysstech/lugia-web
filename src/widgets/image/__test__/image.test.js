/*
 *@flow
 *
 */

import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Image from '../index';

import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

Enzyme.configure({ adapter: new Adapter() });
describe('Image', () => {
  it('image: isBackground = false', () => {
    const target = renderer
      .create(
        <Image
          src="https://interactive-examples.mdn.mozilla.net/media/examples/grapefruit-slice-332-332.jpg"
          alt="我是可爱的西柚"
          title="I am lovely grapefruit"
          height="200"
          width="250"
        />
      )
      .toJSON();
    expect(target).toMatchSnapshot();
  });

  it('background: isBackground = true', () => {
    const target = renderer
      .create(
        <Image
          src="https://interactive-examples.mdn.mozilla.net/media/examples/grapefruit-slice-332-332.jpg"
          alt="我是可爱的西柚"
          title="I am lovely grapefruit"
          height="200"
          width="250"
          isBackground={true}
        />
      )
      .toJSON();
    expect(target).toMatchSnapshot();
  });
});
