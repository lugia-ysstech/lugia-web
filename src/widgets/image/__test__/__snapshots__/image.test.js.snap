// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image background: isBackground = true 1`] = `
.c0 {
  box-sizing: content-box;
  width: 20rem;
  height: 10rem;
  background-image: url(https://interactive-examples.mdn.mozilla.net/media/examples/grapefruit-slice-332-332.jpg);
}

<div
  className="c0"
  style={Object {}}
  title="I am lovely grapefruit"
/>
`;

exports[`Image image: isBackground = false 1`] = `
.c0 {
  width: 20rem;
}

<img
  alt="我是可爱的西柚"
  className="c0"
  src="https://interactive-examples.mdn.mozilla.net/media/examples/grapefruit-slice-332-332.jpg"
  style={Object {}}
  title="I am lovely grapefruit"
/>
`;
